apps/custom_auth/forms.py

```diff
@@ -0,0 +1,132 @@
+import xmltodict
+from allauth.account.forms import LoginForm
+from allauth.account.forms import SignupForm
+from django import forms
+from django.conf import settings
+from django.core.exceptions import ObjectDoesNotExist
+from django.utils.translation import get_language
+from django.utils.translation import ugettext_lazy as _
+from zeep import Client
+
+from apps.organisations.models import Member
+from apps.organisations.models import Organisation
+
+
+class DefaultLoginForm(LoginForm):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['login'].label = _('Username/e-mail')
+        del self.fields['login'].widget.attrs['placeholder']
+        del self.fields['password'].widget.attrs['placeholder']
+
+
+class DefaultSignupForm(SignupForm):
+    terms_of_use = forms.BooleanField(
+        label=_('Terms of use')
+    )
+    get_newsletters = forms.BooleanField(
+        label=_('Info Mails'),
+        help_text=_('Projects I follow are allowed to send me '
+                    'information mails.'),
+        required=False
+    )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['username'].help_text = \
+            _("Your username will appear publicly next to your posts.")
+        del self.fields['username'].widget.attrs['autofocus']
+        self.fields['email'].widget.attrs['autofocus'] = True
+        del self.fields['username'].widget.attrs['placeholder']
+        del self.fields['email'].widget.attrs['placeholder']
+        del self.fields['password1'].widget.attrs['placeholder']
+        del self.fields['password2'].widget.attrs['placeholder']
+
+    def save(self, request):
+        user = super(DefaultSignupForm, self).save(request)
+        user.get_newsletters = self.cleaned_data['get_newsletters']
+        user.language = get_language()
+        user.save()
+        return user
+
+
+class IgbceSignupForm(DefaultSignupForm):
+    member_number = forms.IntegerField(
+        label=_('IG-BCE member number'),
+        help_text=_('Some helptext.'),
+        max_value=99999999999999999999,
+        min_value=0
+    )
+    birth_date = forms.DateField(
+        label=_('Birth date'),
+        help_text=_('Some helptext.')
+    )
+
+    def validateMemberNumberAndDate(self, member_number, birth_date):
+
+        if (not hasattr(settings, 'IGBCE_NAV_URL') or not hasattr(
+                settings, 'IGBCE_NAV_SECURITYID')):
+            raise forms.ValidationError(
+                "Something is wrong with the setup - please try again later"
+            )
+
+        try:
+            Member.objects.get(member_number=member_number)
+            raise forms.ValidationError(
+                "A member with that number already exists."
+            )
+        except ObjectDoesNotExist:
```

Comment:
I know, I don't strictly need the except, but having just the try makes me cringe.

Reply:
`Member.objects.filter(member_number=member_number).exists()` should be cleaner

Reply:
And, to add more smartassing: `Member.DoesNotExist` ;)

Reply:
Ah! That's the way!